docs/dev/sql-nested-function-select-clause.md

```diff
@@ -0,0 +1,413 @@
+                                            "includes": [
+                                                "comment.data"
+                                            ],
+                                            "excludes": []
+                                        }
+                                    }
+                                }
+                            },
+                            {
+                                "nested": {
+                                    "query": {
+                                        "match_all": {
+                                            "boost": 1.0
+                                        }
+                                    },
+                                    "path": "message",
+                                    "ignore_unmapped": false,
+                                    "score_mode": "none",
+                                    "boost": 1.0,
+                                    "inner_hits": {
+                                        "ignore_unmapped": false,
+                                        "from": 0,
+                                        "size": 3,
+                                        "version": false,
+                                        "seq_no_primary_term": false,
+                                        "explain": false,
+                                        "track_scores": false,
+                                        "_source": {
+                                            "includes": [
+                                                "message.info"
+                                            ],
+                                            "excludes": []
+                                        }
+                                    }
+                                }
+                            }
+                        ],
+                        "adjust_pure_negative": true,
+                        "boost": 1.0
+                    }
+                }
+            ],
+            "adjust_pure_negative": true,
+            "boost": 1.0
+        }
+    },
+    "_source": {
+        "includes": [
+            "message.info",
+            "comment",
+            "comment.data",
+            "message"
+        ],
+        "excludes": []
+    }
+}
+```
+
+## 2 Architecture Diagrams
+
+### 2.1 Composite States for Nested Query Execution
+
+Nested function state diagram illustrating states in SQL plugin for push down execution. The nested operator stays in the `PhysicalPlan` after push down for flattening operation in post-processing. See section [2.3](#24-select-clause-nested-query-class-diagram) for flattening sequence and description.
+
+```mermaid
+stateDiagram-v2
+direction LR
+    LogicalPlan --> OptimizedLogicalPlan: Optimize
+    OptimizedLogicalPlan --> PhysicalPlan:push down
+    note right of PhysicalPlan
+    NestedOperator stays in PhysicalPlan\nafter push down for post-processing.
```

Comment:
Is the \n supposed to be there? It appears to be a newline character but considering I don't see any other \n throughout the rest of the document I think it might be accidental.